src/biokbase/narrative/jobs/jobmanager.py

```diff
@@ -170,33 +234,40 @@ def initialize_jobs(self, cell_ids: List[str] = None) -> None:
 
             self.register_new_job(job, refresh)
 
-    def _create_jobs(self, job_ids) -> dict:
+    def _create_jobs(self, job_ids: List[str]) -> dict:
         """
+        Given a list of job IDs, creates job objects for them and populates the _running_jobs dictionary.
         TODO: error handling
-        Given a list of job IDs, creates job objects for them and populates the _running_jobs dictionary
+
+        :param job_ids: job IDs to create job objects for
+        :type job_ids: List[str]
+
+        :return: dictionary of job states indexed by job ID
+        :rtype: dict
         """
         job_ids = [job_id for job_id in job_ids if job_id not in self._running_jobs]
         if not len(job_ids):
             return {}
 
-        job_states = clients.get("execution_engine2").check_jobs(
-            {
-                "job_ids": job_ids,
-                "exclude_fields": JOB_INIT_EXCLUDED_JOB_STATE_FIELDS,
-                "return_list": 0,
-            }
-        )
+        job_states = Job.query_ee2_states(job_ids, init=True)
```

Comment:
Refactoring the `ee2.check_jobs` mainly because that's another way to increase the psychological chokepoint-ness of updated `job._acc_state` ... at least in the sense that it DRYs things up and reduces cognitive load

Reply:
Sensible. Ideally all of the ee2 methods would be in one place, but that's probably a refactor too far...